### ISSUE TITLE
Switching from one animation to another (like in documentation) fails

### ISSUE THREAD

**Issue by warang580** [replit/kaboom#34]
I'm trying to make animations work but switching animations like this fails :

(heart spawns animation once, then idle animation looping)

![image](https://user-images.githubusercontent.com/233165/115992827-eabcff00-a5cf-11eb-8cc5-61b5af82feaf.png)

It seems like .play() inside the ballback tries to end current animations and trigger onAnimEnd callbacks, but I'm already in one so it makes an infinite loop.

**Comment by warang580** [comment]
Ok I fixed it:

Instead of
```js
stop() {
	if (!curAnim) {
		return;
	}
	
	if (events[curAnim]?.end) {
		events[curAnim].end();
	}
	curAnim = null;
},
```
You need

```js
stop() {
	if (!curAnim) {
		return;
	}

	const callback = events[curAnim]?.end;

        // reset animation before callback so it can change animation
	curAnim = null;

	if (callback) {
		callback();
	}
},
```

**Comment by slmjkdbtl** [comment]
Nice catch! Fixed in f24647c
